src/metadata/metadata.controller.ts

```diff
@@ -0,0 +1,186 @@
+  })
+  @ApiParam({
+    name: 'ticker',
+    description: 'The ticker of the Asset whose metadata is to be fetched',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @ApiParam({
+    name: 'type',
+    description: 'The type of Asset Metadata to be filtered',
+    enum: MetadataType,
+    example: MetadataType.Local,
+  })
+  @ApiParam({
+    name: 'id',
+    description: 'The ID of Asset Metadata to be filtered',
+    type: 'string',
+    example: '1',
+  })
+  @ApiOkResponse({
+    description: 'Details of an Asset Metadata including name, specs and value',
+    type: MetadataDetailsModel,
+  })
+  @Get(':type/:id')
+  public async getSingleMetadata(
+    @Param() params: MetadataParamsDto
+  ): Promise<MetadataDetailsModel> {
+    const metadataEntry = await this.metadataService.findOne(params);
+
+    return createMetadataDetailsModel(metadataEntry);
+  }
+
+  @ApiOperation({
+    summary: 'Create a local metadata for an Asset and optionally set its value.',
+    description:
+      'This endpoint creates a local metadata for the given Asset. The metadata value can be set by passing `value` parameter and specifying other optional `details` about the value',
+  })
+  @ApiParam({
+    name: 'ticker',
+    description: 'The ticker of the Asset for which the metadata is to be created',
+    type: 'string',
+    example: 'TICKER',
+  })
+  @ApiTransactionResponse({
+    description: 'Newly created Metadata entry along with transaction details',
+    type: CreatedMetadataEntryModel,
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.NOT_FOUND]: ['The Asset was not found'],
+    [HttpStatus.BAD_REQUEST]: [
+      'Asset Metadata name length exceeded',
+      'Asset Metadata value length exceeded',
+    ],
+    [HttpStatus.UNPROCESSABLE_ENTITY]: [
+      'Asset Metadata with the given name already exists',
+      'Locked until date of the Metadata value must be in the future',
+      'Expiry date for the Metadata value must be in the future',
+    ],
+  })
+  @Post('create')
+  public async createMetadata(
+    @Param() { ticker }: TickerParamsDto,
+    @Body() params: CreateMetadataDto
+  ): Promise<TransactionResponseModel> {
+    const serviceResult = await this.metadataService.create(ticker, params);
+
+    const resolver: TransactionResolver<MetadataEntry> = ({
+      transactions,
+      result: {
+        asset: { ticker: assetTicker },
+        id,
+        type,
+      },
+    }) =>
+      new CreatedMetadataEntryModel({
+        transactions,
+        metadata: new MetadataEntryModel({ asset: assetTicker, id, type }),
+      });
+
+    return handleServiceResult(serviceResult, resolver);
+  }
+
+  @ApiOperation({
+    summary: 'Set Asset Metadata value and/or its details',
+    description:
+      'This endpoint assigns a new value for the Metadata along with its details or optionally only set the details (expiry + lock status) of the Metadata value. Note that the value of a locked Metadata cannot be altered',
```

Comment:
```suggestion
      'This endpoint assigns a new value for the Metadata along with its expiry and lock status (when provided with `details`) of the Metadata value. Note that the value of a locked Metadata cannot be altered',
```

Reply:
I have changed the text, but to be clear what we can do with this API is that we can 
* only set value
* set value along with details
* only set details (this requires the value to be already set)